File: Medium/189_RotateArray.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Void} Do not return anything, modify nums in-place instead.
def rotate(nums, k)
    count = 0
    start = 0
    while (count < nums.size)
        current = start
        prev = nums[start]
        loop do
            next_index = (current + k) % nums.size;
            temp = nums[next_index]
            nums[next_index] = prev
            prev = temp;
            current = next_index
            count += 1
            break if start == current
        end
        start += 1
    end
end

File: Medium/189_RotateArray.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        count = 0
        start = 0
        while (count < len(nums)):
            current = start
            prev = nums[start]
            while(True):
                next = (current + k) % len(nums)
                temp = nums[next]
                nums[next] = prev
                prev = temp
                current = next
                count += 1
                if start == current:
                    break        
            start += 1
